cmd/di.go

```diff
@@ -244,11 +251,24 @@ func (di *Dependencies) bootstrapNodeComponents(nodeOptions node.Options) {
 		return dialogEstablisher.EstablishDialog(providerID, contact)
 	}
 
-	promiseIssuerFactory := func(issuerID identity.Identity, dialog communication.Dialog) connection.PromiseIssuer {
-		if nodeOptions.ExperimentPromiseCheck {
-			return promise_noop.NewPromiseIssuer(issuerID, dialog, di.SignerFactory(issuerID))
+	paymentIssuerFactory := func(
```

Comment:
We should have a separate NewPaymentIssuerFactory() method, not to clutter tidy bootstrapNodeComponents() function.

Reply:
moved